dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/udp/UDPUtils.java

```diff
@@ -0,0 +1,253 @@
+     * 
+     * @param framework
+     * @param options
+     * @return
+     * @throws NettyException
+     */
+    public static BootstrapExtended createUDPBootstrap(NettyFrameworkImpl framework, Map<String, Object> options)
+            throws NettyException {
+        return create(framework, new UDPConfigurationImpl(options, true));
+    }
+
+    /**
+     * Create a {@link BootstrapExtended} for outbound channels
+     * 
+     * @param framework
+     * @param options
+     * @return
+     * @throws NettyException
+     */
+    public static BootstrapExtended createUDPBootstrapOutbound(NettyFrameworkImpl framework,
+            Map<String, Object> options) throws NettyException {
+        return create(framework, new UDPConfigurationImpl(options, false));
+    }
+
+    private static BootstrapExtended create(NettyFrameworkImpl framework, UDPConfigurationImpl config) {
+        BootstrapExtended bs = new BootstrapExtended();
+        bs.applyConfiguration(config);
+        bs.group(framework.getParentGroup());
+        bs.channel(NioDatagramChannel.class);
+        return bs;
+    }
+
+    private static ChannelFuture bind(NettyFrameworkImpl framework, BootstrapExtended bootstrap, String inetHost,
+            int inetPort, final int retryCount, final int retryDelay, ChannelFutureListener bindListener) {
+        ChannelFuture bindFuture = bootstrap.bind(inetHost, inetPort);
+        if (inetHost.equals("*")) {
+            inetHost = "0.0.0.0";
+        }
+        final String newHost = inetHost;
+
+        if (bindListener != null) {
+            bindFuture.addListener(bindListener);
+        }
+        final UDPConfigurationImpl config = ((UDPConfigurationImpl) bootstrap.getConfiguration());
+        final String channelName = ((UDPConfigurationImpl) bootstrap.getConfiguration()).getExternalName();
+
+        bindFuture.addListener(future -> {
+            if (future.isSuccess()) {
+
+                // add the new channel to the set of active channels, and set a close future to
+                // remove it
+                final Channel channel = bindFuture.channel();
+                framework.getActiveChannels().add(bindFuture.channel());
+                channel.closeFuture().addListener(innerFuture -> framework.stop(channel));
+
+                // set common channel attrs
+                channel.attr(ConfigConstants.NameKey).set(config.getExternalName());
+                channel.attr(ConfigConstants.HostKey).set(newHost);
+                channel.attr(ConfigConstants.PortKey).set(inetPort);
+                channel.attr(ConfigConstants.IsInboundKey).set(config.isInboundChannel());
+                
+                // set up the a helpful log message
+                String hostLogString = newHost == "0.0.0.0" ? "*" : newHost;
+                SocketAddress addr = channel.localAddress();
+                InetSocketAddress inetAddr = (InetSocketAddress)addr;
+                String IPvType = "IPv4";
```

Comment:
constant, same for IPv6

Reply:
Actually this is the only place these string values are used.